urlfcc351a1ba_kuntapavankumar_Z_tgz-pJ8-edu_concordia_dpis_ReplicaTestJ8/src/edu/concordia/dpis/MulticastReceiver.java

```diff
@@ -0,0 +1,55 @@
+package edu.concordia.dpis;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.net.SocketException;
+
+@Deprecated
+public class MulticastReceiver extends Thread{
+	
+	private int port;
+	
+	public MulticastReceiver(int port) {
+		this.port = port;
+	}
+	
+	public void run(){
+		
+		MulticastSocket socket = null;
+		DatagramPacket inPacket = null;
+		DatagramSocket socket2 = null;
+		byte[] receiveData = new byte[1024];
+		
+		try{
+			System.out.println("Receiver is up");
+			socket = new MulticastSocket(port);
+			InetAddress address = InetAddress.getByName("230.0.0.1");
+			socket.joinGroup(address);
+			
+			while(true){
+				inPacket = new DatagramPacket(receiveData, receiveData.length);
+				socket.receive(inPacket);
+				
+				String received = new String(inPacket.getData());
+				System.out.println(received);
+				
+				// after processing the request when you have to send the result back
+				socket2 = new DatagramSocket();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *socket2*. Currently, there are execution paths that do not contain closure statements, for example, when DatagramSocket.send throws an exception. Ensure *socket2* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).